crates/league-mod/src/utils/league_path.rs

```diff
@@ -0,0 +1,199 @@
+//! League of Legends path detection and validation utilities.
+
+use camino::Utf8PathBuf;
+use std::fs;
+use std::path::Path;
+use sysinfo::{Disks, System};
+
+/// Validates if a path points to a valid League of Legends executable.
+pub fn is_valid_league_path(path: &str) -> bool {
+    let p = Path::new(path);
+    if !p.exists() {
+        return false;
+    }
+    if let Some(file_name) = p.file_name() {
+        return file_name == "League of Legends.exe";
+    }
+    false
+}
+
+/// Get all available drives using sysinfo (cross-platform).
+fn get_available_drives() -> Vec<String> {
+    let disks = Disks::new_with_refreshed_list();
+
+    let mut drives: Vec<String> = disks
+        .iter()
+        .filter_map(|disk| disk.mount_point().to_str().map(|s| s.to_string()))
+        .collect();
+
+    // Fallback to common Windows drives if detection fails
+    if drives.is_empty() && cfg!(target_os = "windows") {
+        drives = vec!["C:", "D:", "E:", "F:", "G:", "H:"]
+            .into_iter()
+            .map(String::from)
+            .collect();
+    }
+
+    drives
+}
+
+/// Detect League installation from RiotClientInstalls.json.
+fn detect_from_riot_client_installs() -> Option<String> {
+    // Build path: C:\ProgramData\Riot Games\RiotClientInstalls.json
+    let system_drive = std::env::var("SystemDrive").unwrap_or_else(|_| "C:".to_string());
+    let system_root = format!("{}\\", system_drive);
+
+    let riot_installs_path = Utf8PathBuf::from(&system_root)
+        .join("ProgramData")
+        .join("Riot Games")
+        .join("RiotClientInstalls.json");
+
+    if !Path::new(riot_installs_path.as_str()).exists() {
+        return None;
+    }
+
+    let contents = fs::read_to_string(riot_installs_path.as_str()).ok()?;
+    let data: serde_json::Value = serde_json::from_str(&contents).ok()?;
+    let associated_client = data.get("associated_client")?.as_object()?;
+
+    for (install_path, _) in associated_client {
+        // Normalize: remove trailing slash and convert to backslashes
+        let cleaned_path = install_path.trim_end_matches('/').replace('/', "\\");
```

Comment:
I'm pretty sure this is redundant since we create a PathBuf in the line afterwards, that should handle the path separator uncertainty